### PR TITLE
FocusZone: Fix memory leaks when first and/or focused elements are removed

### LINKED ISSUE #23361: [Bug]: Memory leak on FocusZone
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.13 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
    Memory: 33.63 GB / 62.81 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
```

### Are you reporting Accessibility issue?

no

### Reproduction

Provided on the description

### Bug Description

This leak was found during an audit in OWA — Outlook Web. Specifically, while looking for leaks on `<Ribbon/>` component.

## Reproduction steps

1. Navigate to https://outlook-sdf.office.com/mail
1. F12, go to Detached Elements
2. Click on "New Mail"
3. Click "Discard" to delete the New Mail
4. Note, on Detached Elements, that we're leaking two objects, both of which are an element with `data-unique-id=RibbonBottomBarContainer`. One leak was found to be within 1JS/acui-ribbon-like boundaries; the second one was found to be leaking on FocusZone.

## Investigation

The leaking snippet is on `FocusZone.tsx:169`, which happens to be the class' constructor method (on minified code). Upon debugging, I've found that the object's context is being kept in a ref box through in `initializeComponentRef`'s `_onUpdate` (@fluentui/utilities/initializeComponentRef.js:20) method. Which I'd say is the mostly likely cause to keep the HTML ref alive across renderings – even though the `_activeElement` attribute is being disposed on `componentWillUnmount` (FocusZone.tsx:283).

 I couldn't inspect further than this because it is happening on a dependency (@fluentui/react-utilities) of a dependency (@fluentui/react-focus) of a dependency (@1js/acui-ribbon-like).

df5d159a-ab67-4b66-9ea5-be613f32ccc1
dc759e88-1d2f-4daf-b926-ffd732c959f0

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

OWA

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by jspurlin**
## Current Behavior

The `FocusZone`'s `_defaultFocusElement` and `_activeElement` members can hold a reference to element that have been removed from the DOM which can lead to memory leaks.

## New Behavior

This PR clears `_defaultFocusElement` and `_activeElement` if either is no longer in the DOM and then calls `updateTabIndexes` to make sure `_activeElement`, `_defaultFocusElement` and the `tab-index`es are updated correctly.

I also added two unit tests for this behavior.

## Related issue(s)

Fixes #23361

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1a38ae7f801177700d8aff9ead4af5263b9638c7'>1a38ae7f801177700d8aff9ead4af5263b9638c7</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/BaseButton-mount_base.html">940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/BaseButton-mount.html">894</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Breadcrumb-mount_base.html">2885</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Breadcrumb-mount.html">2897</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Checkbox-mount_base.html">1591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Checkbox-mount.html">1622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/CheckboxBase-mount_base.html">1323</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/CheckboxBase-mount.html">1292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ChoiceGroup-mount_base.html">4869</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ChoiceGroup-mount.html">4875</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ComboBox-mount_base.html">1004</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ComboBox-mount.html">986</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/CommandBar-mount_base.html">10922</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/CommandBar-mount.html">11270</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ContextualMenu-mount_base.html">12191</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ContextualMenu-mount.html">12091</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DefaultButton-mount_base.html">1212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DefaultButton-mount.html">1172</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRow-mount_base.html">3952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRow-mount.html">3846</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRowFast-mount_base.html">3963</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRowFast-mount.html">3979</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRowNoStyles-mount_base.html">3784</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DetailsRowNoStyles-mount.html">3733</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Dialog-mount_base.html">2340</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Dialog-mount.html">2304</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DocumentCardTitle-mount_base.html">182</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/DocumentCardTitle-mount.html">169</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Dropdown-mount_base.html">3419</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Dropdown-mount.html">3383</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/FocusTrapZone-mount_base.html">1872</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/FocusTrapZone-mount.html">1884</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/FocusZone-mount_base.html">1901</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/FocusZone-mount.html">1896</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/IconButton-mount_base.html">1807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/IconButton-mount.html">1801</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Label-mount_base.html">365</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Label-mount.html">365</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Layer-mount_base.html">3037</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Layer-mount.html">3058</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Link-mount_base.html">475</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Link-mount.html">524</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/MenuButton-mount_base.html">1574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/MenuButton-mount.html">1594</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/MessageBar-mount_base.html">2332</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/MessageBar-mount.html">2161</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Nav-mount_base.html">3368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Nav-mount.html">3411</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/OverflowSet-mount_base.html">1126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/OverflowSet-mount.html">1149</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Panel-mount_base.html">2318</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Panel-mount.html">2214</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Persona-mount_base.html">1031</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Persona-mount.html">1032</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Pivot-mount_base.html">1441</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Pivot-mount.html">1518</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/PrimaryButton-mount_base.html">1349</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/PrimaryButton-mount.html">1360</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Rating-mount_base.html">7889</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Rating-mount.html">7853</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SearchBox-mount_base.html">1319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SearchBox-mount.html">1318</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Shimmer-mount_base.html">2585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Shimmer-mount.html">2589</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Slider-mount_base.html">1994</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Slider-mount.html">1933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SpinButton-mount_base.html">5087</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SpinButton-mount.html">5155</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Spinner-mount_base.html">446</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Spinner-mount.html">442</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SplitButton-mount_base.html">3316</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SplitButton-mount.html">3263</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Stack-mount_base.html">519</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Stack-mount.html">530</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/StackWithIntrinsicChildren-mount_base.html">2346</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/StackWithIntrinsicChildren-mount.html">2320</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/StackWithTextChildren-mount_base.html">5348</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/StackWithTextChildren-mount.html">5420</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SwatchColorPicker-mount_base.html">11888</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/SwatchColorPicker-mount.html">11948</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TagPicker-mount_base.html">2780</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TagPicker-mount.html">2774</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TeachingBubble-mount_base.html">98655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TeachingBubble-mount.html">98222</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Text-mount_base.html">433</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Text-mount.html">423</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TextField-mount_base.html">1476</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/TextField-mount.html">1443</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-mount_base.html">1199</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-mount.html">1236</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-virtual-rerender_base.html">650</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-virtual-rerender.html">660</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1971</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1970</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Toggle-mount_base.html">811</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/Toggle-mount.html">822</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/buttonNative-mount_base.html">134</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23445/perf-test/buttonNative-mount.html">131</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Comment by ruiconti** [comment]
#23361

File: packages/react-focus/src/components/FocusZone/FocusZone.tsx [hunk 1]
```diff
@@ -243,6 +243,18 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
...
+    // reset those variables (and update tab indexes) to avoid memory leaks
+    if (
+      (this._activeElement && !elementContains(this._root.current, this._activeElement, ALLOW_VIRTUAL_ELEMENTS)) ||
+      (this._defaultFocusElement &&
+        !elementContains(this._root.current, this._defaultFocusElement, ALLOW_VIRTUAL_ELEMENTS))
+    ) {
+      this._activeElement = null;
+      this._defaultFocusElement = null;
+      this._updateTabIndexes();
+    }
```
Comment (khmakoto) [inline packages/react-focus/src/components/FocusZone/FocusZone.tsx:256]: Should we reset both whenever either of them is removed or should there be a separate if statement for each of them (with maybe a boolean flag to indicate if one of them is reset so tab indexes are updated only once)?

Comment (jspurlin) [reply packages/react-focus/src/components/FocusZone/FocusZone.tsx:256]: They could be updated if either of them are removed (in `updateTabIndexes`). This helps ensure that either of them aren't going to leak, in the case where one was null, the other points to a detached element, but `updateTabIndexes` doesn't update the variable holding on to the detached element.

It's more of a "safe extra-precaution"

**Review by khmakoto — APPROVED** [review]

**Pull request merged**